### ISSUE TITLE
Manual panic triggers for `cfg!`

### ISSUE THREAD

**Issue by dev-ardi** [rust-lang/rust-clippy#13883]
### Summary

This is a fairly (?) common idiom, [so much that it even has its own crate](https://docs.rs/debug_panic/latest/debug_panic/macro.debug_panic.html).

### Reproducer

I tried this code:

```rust
<code>
```

I saw this happen:

```
<output>
```

I expected to see this happen:

### Version

```text

```

**Comment by alex-semenyuk** [comment]
I don't see that it triggers now https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c33d342d9d8aafce71d93bf3f2346b4a

**Comment by ada4a** [comment]
It still does, unfortunately -- you just need to explicitly `#![warn(clippy::manual_assert)]`, since the lint is allow-by-default.

Looking into it

**Comment by ada4a** [comment]
Closing as per https://github.com/rust-lang/rust-clippy/pull/15651#pullrequestreview-3221875287
